amxmodx/newmenus.cpp

```diff
@@ -1076,6 +1076,112 @@ static cell AMX_NATIVE_CALL menu_setprop(AMX *amx, cell *params)
 	return 1;
 }
 
+// Gets a menu property
+// native menu_getprop(menu, prop, ...);
+static cell AMX_NATIVE_CALL menu_getprop(AMX *amx, cell *params)
+{
+	enum args { arg_numargs, arg_menu, arg_prop, arg_buffer, arg_bufferlen };
+
+	GETMENU(params[arg_menu]);
+
+	int paramsNum = params[arg_numargs] / sizeof(cell);
+
+	switch(params[arg_prop])
+	{
+		case MPROP_PAGE_CALLBACK: return pMenu->pageCallback;
+		case MPROP_SHOWPAGE: return pMenu->showPageNumber;
+		case MPROP_SET_NUMBER_COLOR:
+		{
+			if (paramsNum < arg_bufferlen)
+			{
+				LogError(amx, AMX_ERR_NATIVE, "Expected 4 parameters");
+				return 0;
+			}
+
+			set_amxstring(amx, params[arg_buffer], pMenu->m_ItemColor.chars(), params[arg_bufferlen]);
+			break;
+		}
+		case MPROP_PERPAGE:	return pMenu->items_per_page;
+		case MPROP_BACKNAME:
+		{
+			if (paramsNum < arg_bufferlen)
+			{
+				LogError(amx, AMX_ERR_NATIVE, "Expected 4 parameters");
+				return 0;
+			}
+
+			set_amxstring(amx, params[arg_buffer], pMenu->m_OptNames[abs(MENU_BACK)].chars(), params[arg_bufferlen]);
+			break;
+		}
+		case MPROP_NEXTNAME:
+		{
+			if (paramsNum < arg_bufferlen)
+			{
+				LogError(amx, AMX_ERR_NATIVE, "Expected 4 parameters");
+				return 0;
+			}
+
+			set_amxstring(amx, params[arg_buffer], pMenu->m_OptNames[abs(MENU_MORE)].chars(), params[arg_bufferlen]);
+			break;
+		}
+		case MPROP_EXITNAME:
+		{
+			if (paramsNum < arg_bufferlen)
+			{
+				LogError(amx, AMX_ERR_NATIVE, "Expected 4 parameters");
+				return 0;
+			}
+
+			set_amxstring(amx, params[arg_buffer], pMenu->m_OptNames[abs(MENU_EXIT)].chars(), params[arg_bufferlen]);
+			break;
+		}
+		case MPROP_TITLE:
+		{
+			if (paramsNum < arg_bufferlen)
+			{
+				LogError(amx, AMX_ERR_NATIVE, "Expected 4 parameters");
+				return 0;
+			}
+
+			set_amxstring(amx, params[arg_buffer], pMenu->m_Title.chars(), params[arg_bufferlen]);
+			break;
+		}
+		case MPROP_EXITALL:
+		{
+			if(pMenu->m_NeverExit == false && pMenu->m_ForceExit == false)
+			{
+				return 1;
+			}
+			else if(pMenu->m_NeverExit == false && pMenu->m_ForceExit == true)
+			{
+				return 2;
+			}
+			else if(pMenu->m_NeverExit == true && pMenu->m_ForceExit == false)
+			{
+				return -1;
+			}
+			
+			return 0;
+		}
+		case MPROP_ORDER:
+		{
+			/* Ignored as of 1.8.0 */
+			break;
+		}
+		case MPROP_NOCOLORS:
+		{
+			return pMenu->m_AutoColors;
```

Comment:
Here should be explicit conversion from `bool` to `cell`. Moreover it should be inversed, see `menu_setprop` code.

Reply:
Just to make it clear - use `static_cast<T>` instead of C `(T)` casts please.

Reply:
Done.

Reply:
@OciXCrom you forgot about inversing.

Reply:
It should be `static_cast<cell>(pMenu->m_AutoColors ? false : true)`, right?

Reply:
This can be either: `pMenu->m_AutoColors ? FALSE : TRUE` or `static_cast<cell>(!pMenu->m_AutoColors)` I guess.

Reply:
Done.

Reply:
@OciXCrom `static_cast<cell>(!pMenu->m_AutoColors)`.